### PR TITLE
Pass Itemset selection by keys 

### PR DESCRIPTION
**Pull request by shubham1g5**
## Technical Summary

[Spec](https://docs.google.com/document/d/1Xtly4YfrdaB1w1Ozwlwy9LQ4JIrVRmhvg-44-Rl-z6A/edit#)

### Automated test coverage

Added tests for new bheaviour while retaining tests for old behaviour. 

### QA Plan

[Outlined here](https://dimagi-dev.atlassian.net/browse/QA-4993?focusedCommentId=260146)

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code.

### Special deploy instructions

- This can only be deployed after the related Web Apps PRs have been deployed to ensure backward compatibility. 

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations.

### Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change.

core - https://github.com/dimagi/commcare-core/pull/1261

### CODE REVIEW

**Review by snopoke — APPROVED** [review]
Confirm that with these changes FP will still accept the old request format but will always respond with the new format.

File: src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java [hunk 1]
```diff
@@ -54,27 +55,11 @@ public QueryResponseBean(QueryScreen queryScreen, SessionWrapper session) {
...
-                            indicesForSelectedChoices.add(String.valueOf(choiceIndex));
-                        }
-                    }
-                }
-                if (indicesForSelectedChoices.size() > 0) {
-                    currentAnswer = String.join(RemoteQuerySessionManager.ANSWER_DELIMITER,
-                            indicesForSelectedChoices);
-                }
-                choiceLabels = ItemSetUtils.getChoiceLabels(queryPromptItem.getItemsetBinding());
+                Pair<String[], String[]> choices = ItemSetUtils.getChoices(queryPromptItem.getItemsetBinding());
```
Comment (Jtang-1) [inline src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java:60]: Is this a real test failure? getChoices doesn't seem to be defined

Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java:60]: Oh nvm i see this was in commcare-core PR

File: src/main/java/org/commcare/formplayer/beans/menus/DisplayElement.java [hunk 2]
```diff
@@ -154,6 +159,11 @@ public String[] getItemsetChoices() {
         return itemsetChoices;
     }
 
+    @Nullable
+    public String[] getItemsetChoiceskey() {
```
Comment (Jtang-1) [inline src/main/java/org/commcare/formplayer/beans/menus/DisplayElement.java:163]:
```suggestion
    public String[] getItemsetChoicesKey() {
```

**Pull request merged**